pyproject.toml

```diff
@@ -39,3 +39,40 @@ skip_glob = [
     "mypyc/test-data/*",
     "test-data/*",
 ]
+
+[tool.ruff]
+line-length = 99
+target-version = "py37"
+
+extend-ignore = [
+  "B006",  # use of mutable defaults in function signatures
+  "B007",  # Loop control variable not used within the loop body.
+  "B011",  # Don't use assert False
+  "B023",  # Function definition does not bind loop variable
+  "E203",  # conflicts with black
+  "E402",  # module level import not at top of file
+  "E501",  # conflicts with black
+  "E731",  # Do not assign a `lambda` expression, use a `def`
+  "E741",  # Ambiguous variable name
+]
+
+extend-exclude = [
+  "@*",
+  # Sphinx configuration is irrelevant
+  "docs/source/conf.py",
+  "mypyc/doc/conf.py",
+  # tests have more relaxed styling requirements
+  # fixtures have their own .pyi-specific configuration
+  "test-data/*",
+  "mypyc/test-data/*",
+  # typeshed has its own .pyi-specific configuration
+  "mypy/typeshed/*",
+]
+
+[tool.ruff.per-file-ignores]
```

Comment:
Why do we need per file ignores here? We should be excluding this entirely, right?

Reply:
I mentioned this when I opened the PR:

> Lastly, the `test-data` folder seemed to have its own `.flake8` config file, but it was disabled in the `setup.cfg` file. I migrated the config over, but it doesn't really matter because it is being ignored anyways. Is this intentional?

So yes, we don't need the `per-file-ignores` since the `test-data` folder is ignored by the `ignore` above. The reason I included it was because of the `.flake8` file in the `test-data` folder. If we don't want to lint the test data folder then we should remove `per-file-ignores`, but if we do want to lint that folder then we should remove `test-data` from the ignore config. I added them both to mimic the flake8 setup that was already there.

Reply:
I went ahead and just remove the `per-file-ignores` since they're just ignored anyways. If we decide to go back and lint the test data files we can re-add the rules we care about.